geomstats/information_geometry/statistical_metric.py

```diff
@@ -0,0 +1,386 @@
+"""Statistical metric and conjugate connections to equip manifold."""
+
+import geomstats.backend as gs
+from geomstats.geometry.connection import Connection
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+
+
+def unpack_inputs(func, dim):
+    r"""Wrap function to unpack inputs to divergence function.
+
+    Compose a function :math:`g:` with a wrapper :math:`f`
+    to identify
+    .. math::
+        g: \mathbb{R}^{n} \times \mathbb{R}^{n} \to \mathbb{R}
+
+    with the function
+    .. math::
+        \tidle{g} = f\circ g: \mathbb{R}^{2n} \to \mathbb{R}
+
+    Autodifferentiation is then computed on :math:`\tidle{g}`.
+
+    Parameters
+    ----------
+    func : callable
+        Function to unpack inputs.
+    dim : int
+        Dimension of the manifold.
+
+    Returns
+    -------
+    callable
+        Composition of wrapper function with original function.
+    """
+    def wrapper(tensor):
+        return func(tensor[..., : dim], tensor[..., dim :])
+    return wrapper
+
+
+class DivergenceConnection(Connection):
+    r"""Class to derive a connection from a divergence.
+
+    When implemented in conjuction with the DualDivergenceConnection object,
+    a pair of conjugate connections can be defined on a manifold.
+
+    Given an :math:`n` manifold :math:`M` with coordinates :math:`x`
+    and a divergence :math:`D: \mathbb{R}^n \times \mathbb{R}^n \to \mathbb{R}`,
+    the conjugate connections :math:`\nabla^D` and :math:`\nabla^{D^*}` can be defined
+    in terms of their corresponding Christoffel symbols :math:`\Gamma^D`
+    and :math:`\Gamma^{D^*}`.
+
+    These connections are said to be conjugate with respect to the divergence induced
+    metric :math:`g^D` because for any smooth vector fields :math:`X, Y, Z`
+    on :math:`M` we have:
+
+    .. math::
+        X(g^D(Y, Z)) = g^D(\nabla^D_X Y, Z) + g^D(Y, \nabla^{D^*}_X Z)
+
+    The Levi-Civita connection can be recovered from these connections:
+
+    .. math::
+        \nabla^{LC} = \frac{\nabla^D + \nabla^{D^*}}{2}
+
+    Attributes
+    ----------
+    space : Manifold
+        Manifold to equip with the divergence and its conjugate connections.
+    divergence : callable
+        Divergence function, takes two inputs of shape=[..., dim] and returns a scalar.
+
+    References
+    ----------
+    .. [N2020] F. Nielsen,
+        "An Elementary Introduction to Information Geometry",
+        arXiv:808.08271v2 (2020): 10-16
+    """
+
+    def __init__(self, space, divergence):
+        super().__init__(space=space)
+        assert (
+            space.intrinsic
+        ), "The manifold must be parametrized by an intrinsic coordinate system."
+        self.dim = space.dim
+        self.divergence = unpack_inputs(divergence, self.dim)
+
+    def christoffels(self, base_point):
```

Comment:
Split into two methods: `first_kind_christoffels` and `christoffels`. Do the same below for the dual (I'll have some questions regarding the dual, but I'll leave them for the next iteration).